code-challenges.js

```diff
@@ -1,13 +1,60 @@
 // ASSESSMENT 5: JavaScript Coding Practical Questions
 
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
-
+// This was very useful!
+// https://stackoverflow.com/questions/1206911/why-do-i-need-to-add-g-when-using-string-replace-in-javascript
+// example
+// let str = 'JS will, JS will rock you!';
+// let newStr = str.replace(/JS/g,'JavaScript');
+
+// Start by creating a function that takes in a string
+// I think there are many many ways to solve this problem. I could use a map or a for loop. I could also do if/else statements or use .replace.
 var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+// Start by creating a function that takes in a string
+const secretMessage = (string) => {
+    // use the .replace method to substitute a for 4, e for 3, i for 1 and o for 0 here
+    return string.replace(/a/g, "4").replace(/e/g, "3").replace(/i/g, "1").replace(/o/g, "0")
+}
+
+// console.log(secretMessage(secretCodeWord1));
+// console.log(secretMessage(secretCodeWord2));
 
+// Even though the function above works, Rachael mentioned that we should use higher order functions whenever we have a chance. 
+// Start by creating a function that takes in a string
+
+var secretCodeWord1 = "lackadaisical"
+// Expected output: "l4ck4d41s1c4l"
+var secretCodeWord2 = "gobbledygook"
+// Expected output: "g0bbl3dyg00k"
+
+// Start by creating a function that takes in a string
+const secretMessageTwo = (string) => {
+// Now create a local variable and use .split in order to split the string and map over it
+    newString = string.split("")
+    // we need to create a second variable in order to use .map
+    var secretString = newString.map(value =>{
+        // If else statements to set letters a/e/i/o to numbers
+        if(value === 'a'){
+            return value = 4
+        }else if(value === 'e'){
+            return value = 3
+        }else if(value === 'i'){
+            return value = 1
+        }else if(value === 'o'){
+            return value = 0
```

Comment:
Consider a way to store the mapping from letter to number without using if statement